src/main/java/com/fasterxml/jackson/databind/jsontype/impl/SimpleClassNameIdResolver.java

```diff
@@ -0,0 +1,194 @@
+package com.fasterxml.jackson.databind.jsontype.impl;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.databind.BeanDescription;
+import com.fasterxml.jackson.databind.DatabindContext;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.cfg.MapperConfig;
+import com.fasterxml.jackson.databind.jsontype.NamedType;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeSet;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * {@link com.fasterxml.jackson.databind.jsontype.TypeIdResolver} implementation
+ * that converts between (JSON) Strings and simple Java class names via {@link Class#getSimpleName()}.
+ * <p>
+ * Note that this implementation is identical to {@link TypeNameIdResolver} except that instead of 
+ * {@link TypeNameIdResolver#_defaultTypeId(Class)}, this implementation uses {@link Class#getSimpleName()}.
+ * 
+ * @since 2.16
+ */
+public class SimpleClassNameIdResolver
+    extends TypeIdResolverBase
+{
+    protected final MapperConfig<?> _config;
+
+    /**
+     * Mappings from class name to type id, used for serialization.
+     *<p>
+     * Since lazily constructed will require synchronization (either internal
+     * by type, or external)
+     */
+    protected final ConcurrentHashMap<String, String> _typeToId;
```

Comment:
I don't think we need to use concurrent data structure when it instances are immutable.
And thereby should use regular `HashMap`.

Reply:
This is not immutable. It has a mutable Map.

I think what is meant is whether a single SimpleClassNameIdResolver instance can be used by 2 threads at the same time. If it can be used by multiple threads, then this should be a concurrent map.

Reply:
@pjfanning Yes you are correct, I first only looked at construction but then noticed mutation later on. So my comment was based on only reading part of the code.